### PR TITLE
Up cloud-provider version

### PR DESCRIPTION
**Pull request by Hitych**
Cloud-provider actual tag now is 0.31.2. There are few impovements in comparison with 0.27.3. Most interesting to me are [metrics](https://github.com/kubernetes/cloud-provider/commit/7d76f758dbe14c08215f39b6f520c51b950a0490) for node-controller. 
It requires minor change in _app.NewCloudControllerManagerCommand_ call related to new controllers naming [convention](https://github.com/kubernetes/cloud-provider/commit/1339cd2f894fd23f470ada0746ccc1ee24e1725d).
I tested base scenario and it works as expected:

1. Create/delete VM
2. Create/delete LoadBalancer service

### CODE REVIEW

**Review by nabokihms — APPROVED** [review]

**Comment by nabokihms** [comment]
@Hitych thanks for the PR. Could you please rebase the branch? I've just updated images in the Dockerfile.

**Comment by Hitych** [comment]
> @Hitych thanks for the PR. Could you please rebase the branch? I've just updated images in the Dockerfile.

done

File: go.mod [hunk 1]
```diff
@@ -2,8 +2,10 @@ module github.com/deckhouse/yandex-cloud-controller-manager
 
 go 1.23
 
+toolchain go1.23.2
```
Comment (nabokihms) [inline go.mod:5]:
```suggestion
```

**Comment by nabokihms** [comment]
@Hitych thank you for your contribution.

**Pull request merged**